### PR TITLE
Add support for backcompat hint flags

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

Add support for backwards-compatibility hint flags.

### API Changes ###

Added:
 - `public static void SetFlags(params string[] flags)` 

### Behavioral Changes ###

Android fast renderers now require setting flag `FastRenderers_Experimental`.

### CODE REVIEW

**Comment by rmarinho** [comment]
build

File: Xamarin.Forms.ControlGallery.Android/FormsAppCompatActivity.cs [hunk 1]
```diff
@@ -42,7 +42,12 @@ protected override void OnCreate(Bundle bundle)
 			if (!Debugger.IsAttached)
 				Insights.Initialize(App.InsightsApiKey, ApplicationContext);
 
+#if TEST_EXPERIMENTAL_RENDERERS
+			Forms.SetFlags("FastRenderers_Experimental");
```
Comment (rmarinho) [inline Xamarin.Forms.ControlGallery.Android/FormsAppCompatActivity.cs]: Shouldn' we use our namespace ?

Comment (hartez) [inline Xamarin.Forms.ControlGallery.Android/FormsAppCompatActivity.cs]: I thought that when we discussed it we decided that our stuff doesn't need namespaces, but we would recommend that third parties using this should prefix with namespaces.

**Comment by bugadani** [comment]
Is there a particular reason flags are basically magic strings? Wouldn't an enum be a better choice? Since the Flags class is internal, we can't use it to have some compiler support for autocompletion and preventing typos, making it annoying to dig through documentation to find the correct flag name and what it does. Also, is there a reason this is being done in runtime, instead of compile time? Coming from an embedded world, wasting code memory when it could be done in an objectively better way bugs me. I'm concerned that this is a rushed change.

**Pull request merged**